src/environments/array.js

```diff
@@ -114,11 +114,8 @@ function parseArray(
             }
             break;
         } else if (next === "\\cr") {
-            const cr = parser.parseFunction();
-            if (!cr) {
-                throw new ParseError(`Failed to parse function after ${next}`);
-            }
-            rowGaps.push(assertNodeType(cr, "cr").size);
+            const cr = assertNodeType(parser.parseFunction(), "cr");
+            rowGaps.push(cr.size);
```

Comment:
I guess this simpler error checking is fine because `parseFunction` will return null only if `\cr` is not a defined function, which shouldn't happen.  Ditto below for `\right`.

Reply:
`assertNodeType` will throw if `parser.parseFunction()` returns null.